lib/typed/success.rb

```diff
@@ -57,5 +57,15 @@ def error
     def and_then(&block)
       block.call(payload)
     end
+
+    sig do
+      override
+        .type_parameters(:Fallback)
+        .params(value: T.type_parameter(:Fallback))
+        .returns(T.any(Payload, T.type_parameter(:Fallback)))
+    end
+    def payload_or(value) # rubocop:disable Lint/UnusedMethodArgument
```

Comment:
nit: you should be able to rename this parameter to `_value` instead of disabling the rubocop rule.
That's normally my preference, but it's not really important 🙂